10_rest/willshen8/pkg/puppy/rest.go

```diff
@@ -0,0 +1,78 @@
+package puppy
+
+import (
+	"net/http"
+	"strconv"
+
+	"github.com/go-chi/chi"
+	"github.com/go-chi/render"
+)
+
+// RestHandler provides a puppy storer to store the puppies
+type RestHandler struct {
+	// sync.Mutex
+	store Storer
+}
+
+// NewRestHandler is a factory method that makes new storer
+func NewRestHandler(storer Storer) *RestHandler {
+	return &RestHandler{
+		store: storer,
+	}
+}
+
+// HandleGet gets the puppy by id and displays the results.
+func (rh *RestHandler) HandleGet(w http.ResponseWriter, r *http.Request) {
+	id, _ := strconv.Atoi(chi.URLParam(r, "id"))
+	puppy, readErr := rh.store.ReadPuppy(uint32(id))
+	if readErr != nil {
+		render.JSON(w, r, readErr)
+	} else {
+		render.JSON(w, r, puppy)
+	}
+}
+
+// HandlePost post a new puppy and then display the results.
+func (rh *RestHandler) HandlePost(w http.ResponseWriter, r *http.Request) {
+	var p Puppy
+	if decodeErr := render.DecodeJSON(r.Body, &p); decodeErr != nil {
+		render.JSON(w, r, decodeErr)
+	}
+	_, createErr := rh.store.CreatePuppy(&p)
```

Comment:
as far as I can see in most cases you are ignoring `bool` returned by crud methods. Maybe there is no sense for returning `bool` from puppy crud?

Reply:
There is definitely a need for it. If I want to make the id into an HTML hyperlink from the response body. Then I can do that. In a real project, I would make use of that and render a better response body, so that user can follow the link and view the results for example, because the user doesn't know the ID being created.